### PR TITLE
shader/validation: Add vector validation tests

### PR DESCRIPTION
**Pull request by ben-clayton**
Fixes: #1441
Fixes: #1610

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/expression/access/vector.spec.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  wxyz_bga_xy: { wgsl: 'let r : vec2<T> = v.wxyz.bga.xy;', ok: (width: number) => width > 3 },
+
+  // error: invalid convenience letterings
+  xq: { wgsl: 'let r : vec2<T> = v.xq;', ok: false },
+  py: { wgsl: 'let r : vec2<T> = v.py;', ok: false },
+
+  // error: mixed convenience letterings
+  xg: { wgsl: 'let r : vec2<T> = v.xg;', ok: false },
+  ryb: { wgsl: 'let r : vec3<T> = v.ryb;', ok: false },
+  xgza: { wgsl: 'let r : vec4<T> = v.xgza;', ok: false },
```
Comment (jrprice) [inline src/webgpu/shader/validation/expression/access/vector.spec.ts:159]: Maybe add one for too many components? e.g. `v.xxxxx`

File: src/webgpu/shader/validation/expression/access/vector.spec.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+alias T = ${t.params.element_type};
+
+struct S {
+  i : i32,
+}
+
+@compute @workgroup_size(1)
+fn main() {
+  var v : vec${t.params.vector_width}<T>;
```
Comment (jrprice) [inline src/webgpu/shader/validation/expression/access/vector.spec.ts:197]: Worth parameterizing the source vector for `var` vs `let/param` vs `const`?

**Review by jrprice — APPROVED** [review]
LGTM with a couple of minor suggestions.

File: src/webgpu/shader/validation/expression/access/vector.spec.ts (hunk 1 above)
Comment (ben-clayton) [reply src/webgpu/shader/validation/expression/access/vector.spec.ts:159]: Done

File: src/webgpu/shader/validation/expression/access/vector.spec.ts (hunk 2 above)
Comment (ben-clayton) [reply src/webgpu/shader/validation/expression/access/vector.spec.ts]: Done

**Pull request merged**